akka-actor-typed/src/main/scala/akka/actor/typed/internal/InterceptorImpl.scala

```diff
@@ -46,16 +50,34 @@ private[akka] final class InterceptorImpl[O, I](val interceptor: BehaviorInterce
   }
 
   private val receiveTarget: ReceiveTarget[I] = new ReceiveTarget[I] {
-    override def apply(ctx: TypedActorContext[_], msg: I): Behavior[I] =
-      Behavior.interpretMessage(nestedBehavior, ctx.asInstanceOf[TypedActorContext[I]], msg)
+    override def apply(ctx: TypedActorContext[_], msg: I): Behavior[I] = {
+      try {
+        val ctxI = ctx.asInstanceOf[TypedActorContext[I]]
+        val nextB = Behavior.interpretMessage(nestedBehavior, ctxI, msg)
+        next(ctxI, nextB)
+      } catch {
+        case NonFatal(ex) ⇒
+          // unstashing is aborted on failure
+          nestedBehavior match {
+            case u: UnstashingBehavior[I] ⇒
+              _nestedBehavior = u.currentBehavior
+            case _ ⇒
+          }
+          throw ex
+
+      }
```

Comment:
This needs to go in signalTarget as well since a user could potentially unstash from a signal?